### PR TITLE
scripts: build-all: warn when ignoring unknown arguments

### PR DESCRIPTION
**Pull request by marc-hb**
Example:
```
 $ ./scripts/xtensa-build-all.sh -j3
 ./scripts/xtensa-build-all.sh: WARN: ignoring arg -j3
```
Signed-off-by: Marc Herbert <marc.herbert@intel.com>

### CODE REVIEW

File: scripts/xtensa-build-all.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ then
 	echo "       [-k] Use private key"
 	echo "       [-v] Verbose Makefile log"
 	echo "       [-j [n]] Set number of make build jobs." \
-		"Jobs=#cores when no flag. Inifinte when no arg."
+		"Jobs=#cores when no flag. No limit when no arg."
```
Comment (paulstelian97) [inline scripts/xtensa-build-all.sh]: I think this should be in a separate commit as it doesn't relate to the commit description.

**Comment by lgirdwood** [comment]
@marc-hb can you easily fix the script with getopt as suggested by @jajanusz ?

**Comment by marc-hb** [comment]
> @marc-hb can you easily fix the script with getopt as suggested by @jajanusz ?

I don't know about "easily" but in terms of time, number of lines changed and risk of not testing some combination or corner case and breaking it, it would be a completely different order of work magnitude. So no, not really.

Totally agree this script shouldn't be re-implementing `getopt` eventually but excluding the typo fix this patch is literally just 3-lines long and adds what is (IMHO) missing the most from this parser.

**Comment by jajanusz** [comment]
Ok, warn is better than nothing, added issue for improvement: https://github.com/thesofproject/sof/issues/2631

File: scripts/xtensa-build-all.sh (hunk 1 above)
Comment (marc-hb) [reply scripts/xtensa-build-all.sh]: I'll drop this typo fix, I don't think it's bad enough to worth a commit alone.

Comment (paulstelian97) [reply scripts/xtensa-build-all.sh]: Maybe good for the next cleanup patch.

Comment (marc-hb) [reply scripts/xtensa-build-all.sh]: Done. Funny enough the checkpatch git commit hook tried to hard to stop me from amending this commit and mispelling Inifinte again :-)

**Comment by lgirdwood** [comment]
Not CI tested for internal CI, Jenkins known issues.

**Comment by marc-hb** [comment]
> I don't know about "easily" but in terms of time, number of lines changed and risk of not testing some combination or corner case and breaking it, it would be a completely different order of work magnitude. So no, not really.

At the time this script only "smelled" bad. Now I know it _is_ bad, see PR #2711
